### PR TITLE
AG 7121 Add double-click detection for event handlers

### PR DESCRIPTION
**Pull request by lsjroberts**
PR for https://ag-grid.atlassian.net/browse/AG-7121

- Adds double click event listeners to the base chart and legend items
- Adds a default behaviour for legend item double clicking that disables all other items except the one double clicked. Double clicking a second time re-enables all items.

### CODE REVIEW

File: charts-packages/ag-charts-community/src/chart/agChartOptions.ts [hunk 1]
```diff
@@ -368,7 +368,20 @@ export interface AgChartLegendItemOptions {
...
+    type: T;
+}
+
+export interface AgChartLegendClickEvent extends AgChartLegendEvent<'click'> {
+    /** Series id */
+    seriesId: string;
+    /** Legend item id - usually yKey value for cartesian series. */
+    itemId: string;
+    /** Whether the legend item is currently enabled or not. */
+    enabled: boolean;
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/chart/agChartOptions.ts:381]: Maybe these properties should live on `AgChartLegendEvent`?

File: charts-packages/ag-charts-community/src/chart/agChartOptions.ts [hunk 2]
```diff
@@ -380,6 +393,8 @@ export interface AgChartLegendClickEvent {
 export interface AgChartLegendListeners {
     /** The listener to call when a legend item is clicked. */
     legendItemClick?: (event: AgChartLegendClickEvent) => void;
+    /** The listener to call when a legend item is double clicked. */
+    legendItemDoubleClick?: (event: AgChartLegendDoubleClickEvent) => void;
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/chart/agChartOptions.ts:397]: Could you check the impact (if any) on Integrated Charts > Cross Filtering API examples? E.g. they support ctrl-click and more advanced toggling / interactions, implemented in Integrated Charts?

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/index.md [hunk 3]
```diff
@@ -71,3 +71,18 @@ This example demonstrates:
 - when a legend item is clicked, an alert message is shown with the `legendItemClick` event contents.
 
 <chart-example title='Legend Item Click Event' name='legend-item-click-event' type='generated'></chart-example>
+
+## Chart Event - click and doubleClick
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/index.md:75]: Should we also add some info to the Legend section about the new double-click behaviour?

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/index.md [hunk 4]
```diff
@@ -71,3 +71,18 @@ This example demonstrates:
...
+
+The `click` and `doubleClick` events are fired when any part of the chart is clicked or double clicked, respectively. The `doubleClick` event does not block the original `click` event from firing.
+
+These events may be prevented by other clickable parts of the chart, such as nodes.
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/index.md:79]: I would clarify `nodes` to `series-nodes`? ...and maybe mention legend items?

File: charts-packages/ag-charts-community/src/chart/legend.ts [hunk 5]
```diff
@@ -754,7 +757,39 @@ export class Legend {
...
+        if (!series) {
+            return;
+        }
+        event.consume();
+
+        if (toggleSeriesVisible) {
+            const legendData = series.getLegendData();
+            const anyVisible = legendData.filter((d) => d.enabled).length;
+            legendData.forEach((d) => {
+                const newEnabled = d.itemId === itemId || anyVisible === 1;
```
Comment (lsjroberts) [inline charts-packages/ag-charts-community/src/chart/legend.ts:785]: This default behaviour doesn't work on line charts, it simply toggles the selected item off and on, the others series are ignored.

File: charts-packages/ag-charts-community/src/chart/agChartOptions.ts (hunk 2 above)
Comment (lsjroberts) [reply charts-packages/ag-charts-community/src/chart/agChartOptions.ts:389]: It doesn't appear to have any impact either way. Double clicking legends doesn't have an effect - I assume click events are handled within integrated charts.

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/examples/legend-item-click-event/main.ts [hunk 6]
```diff
@@ -44,17 +41,26 @@ let options: AgCartesianChartOptions = {
...
+      // },
+      // legendItemDoubleClick: ({
+      //   seriesId,
+      //   itemId,
+      //   enabled,
+      // }: AgChartLegendDoubleClickEvent) => {
+      //   console.log(
+      //     `Double Click - seriesId: ${seriesId}, itemId: ${itemId}, enabled: ${enabled}`
+      //   )
+      // },
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/examples/legend-item-click-event/main.ts:61]: Was this change intentional?

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/index.md [hunk 7]
```diff
@@ -60,14 +60,32 @@ This example demonstrates:
...
 <chart-example title='Node Click Event' name='series-node-click-event' type='generated'></chart-example>
 
-## Legend Event - legendItemClick
+## Legend Event - legendItemClick and legendItemDoubleClick
 
-The `legendItemClick` event can be used to listen to legend item clicks.
+The `legendItemClick` and `legendItemDoubleClick` events can be used to listen to legend item clicks and double clicks, respectively.
 
-### Example: legendItemClick Event
+By default, when a legend item is clicked it hides the related series, and when double clicked it shows that series and hides all others.
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/index.md:67]: The docs here probably need updating too: https://ag-grid.com/javascript-charts/legend/#series-visibility-toggling

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/examples/legend-item-click-event/main.ts (hunk 6 above)
Comment (lsjroberts) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-events/examples/legend-item-click-event/main.ts:61]: doh
